contracts/staking/contracts/src/StakingProxy.sol

```diff
@@ -95,6 +97,36 @@ contract StakingProxy is
         emit ReadOnlyModeSet(readOnlyMode);
     }
 
+    /// @dev Batch executes a series of calls to the exchange contract.
+    /// @param data An array of data that encodes a sequence of functions to
+    ///             call in the staking contracts.
+    function batchExecute(bytes[] memory data)
+        public
+        returns (bytes[] memory batchReturnData)
+    {
+        bool success;
+        bytes memory returnData;
+        batchReturnData = new bytes[](data.length);
+
+        // Execute all of the calls encoded in the provided calldata.
+        for (uint256 i = 0; i < data.length; i++) {
+            // Call the staking contract with the provided calldata.
+            (success, returnData) = stakingContract.simpleProxyCallWithData(data[i]);
```

Comment:
nit: might make sense to just perform the `stakingContract` nil check at the beginning of this function and do the `delegatecall` directly here, otherwise we make that check for each call

Reply:
Hmm, yeah that's a good point. I might just remove `simpleProxyCallWithData in favor of a normal `delegatecall` here.